static/js/util/util.js

```diff
@@ -168,30 +169,58 @@ export function makeCourseProgressDisplay(course, isFirst, isLast) {
  * @param {Object} profile The user profile
  * @returns {Object} Validation errors or an empty object if no errors
  */
-export function validateProfile(profile) {
+export function validateProfile(profile, requiredFields, messages) {
+  let clone = _.cloneDeep(profile);
+  let fields = Object.entries(clone).filter(pair => _.isObject(pair[1]));
+  let nullObj = (obj) => {
+    for (let key in obj) { obj[key] = null };
```

Comment:
Can you change this to `for (let key of Object.keys(obj))` instead? It's slightly more explicit and it guards against possible prototype properties (which are admittedly not a problem for us)